### PR TITLE
#21617 update MySQLDialect for keywords setting

### PR DESCRIPTION
**Pull request by PzaThief**
This PR has whole update of MySQL dialect especially on keywords.
I listed up all keywords of MySQL, So, When initialization of dialect it process clear all default keywords and then register correctly.
In result, there is no more dirty correction like `removeSQLKeyword("SOURCE");`.

But, the problem remains about MySQL version and MySQL-like DB compatibility. (So, I think this patch is just temporary to make user feel at ease.)
Some reserved keywords can be removed always like `NONBLOCKING` ([MySQL Reserved Words](https://dev.mysql.com/doc/mysqld-version-reference/en/keywords.html)) , and MariaDB has another keywords like `OFFSET` ([MariaDB Reserved Words](https://mariadb.com/kb/en/reserved-words/)).

So, I hope to make a disscusion about keeping this design.
I just naively think to make an option(as check-box) to quote all column names whether it is reserved or not.
This approach is used in hibernate (option names hibernate.globally_quoted_identifiers) and I think it not will be hard works.
Could I get your opinion on this?

### CODE REVIEW

**Comment by E1izabeth** [comment]
Thank you for contribution.

File: plugins/org.jkiss.dbeaver.ext.mysql/src/org/jkiss/dbeaver/ext/mysql/model/MySQLDialect.java [hunk 1]
```diff
@@ -43,157 +42,183 @@
...
-        "REPAIR",
-        "ENGINE"
-    };
-
-    public static final String[][] MYSQL_QUOTE_STRINGS = {
-            {"`", "`"},
-            {"\"", "\""},
-    };
-
+            BasicSQLDialect.NON_TRANSACTIONAL_KEYWORDS, new String[] { "USE", "SHOW", "CREATE", "ALTER", "DROP" });
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.ext.mysql/src/org/jkiss/dbeaver/ext/mysql/model/MySQLDialect.java:45]: Why did you delete DESCRIBE and DESC keywords?

Comment (PzaThief) [reply plugins/org.jkiss.dbeaver.ext.mysql/src/org/jkiss/dbeaver/ext/mysql/model/MySQLDialect.java:45]: because BasicSQLDialect.NON_TRANSACTIONAL_KEYWORDS already has DESCRIBE, DESC and EXPLAIN I removed it.

Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.ext.mysql/src/org/jkiss/dbeaver/ext/mysql/model/MySQLDialect.java:45]: Ok, thanks

File: plugins/org.jkiss.dbeaver.ext.mysql/src/org/jkiss/dbeaver/ext/mysql/model/MySQLDialect.java [hunk 2]
```diff
@@ -338,7 +363,8 @@ public boolean validIdentifierStart(char c) {
 
     @NotNull
     @Override
-    public String getTypeCastClause(@NotNull DBSTypedObject attribute, @NotNull String expression, boolean isInCondition) {
+    public String getTypeCastClause(@NotNull DBSTypedObject attribute, @NotNull String expression,
+            boolean isInCondition) {
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.ext.mysql/src/org/jkiss/dbeaver/ext/mysql/model/MySQLDialect.java:367]: Could you revert this unintended change or reformat it?
If the line doesn't fit in 140 characters we format it like this
```suggestion
    public String getTypeCastClause(
        @NotNull DBSTypedObject attribute,
        @NotNull String expression,
        boolean isInCondition
    ) {
```

File: plugins/org.jkiss.dbeaver.ext.mysql/src/org/jkiss/dbeaver/ext/mysql/model/MySQLDialect.java [hunk 3]
```diff
@@ -43,157 +42,183 @@
...
-
-    public static final String[][] MYSQL_QUOTE_STRINGS = {
-            {"`", "`"},
-            {"\"", "\""},
-    };
-
+            BasicSQLDialect.NON_TRANSACTIONAL_KEYWORDS, new String[] { "USE", "SHOW", "CREATE", "ALTER", "DROP" });
+
+    private static final String[] ADVANCED_KEYWORDS = { "AUTO_INCREMENT", "DATABASES", "COLUMNS", "ALGORITHM", "REPAIR",
+            "ENGINE" };
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.ext.mysql/src/org/jkiss/dbeaver/ext/mysql/model/MySQLDialect.java:48]: Please don't change existing formatting like this where it's not needed.

**Comment by E1izabeth** [comment]
Also we usually ask to create an issue on our issue tracker before making a pull request. And you can leave a comment saying that you are going to implement this feature (or fix a bug) by yourself.
https://github.com/dbeaver/dbeaver/wiki/Contribute-your-code

**Comment by E1izabeth** [comment]
@PzaThief could you fix code style and merge conflict?

**Comment by LonwoLonwo** [comment]
Hello.
I checked. MariaDB and MySQL both return the correct colors for the words from the ticket. And for the whole of this list: https://github.com/mysql/mysql-server/blob/mysql-8.2.0/plugin/x/src/mysql_function_names.cc

2024-02-26 11_02_33-Window

So, thanks for the PR, but I do not see any reason to implement it.
